src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -151,8 +170,9 @@ export class ClusterClient {
    * scoped client instances, these will be automatically closed as soon as the
    * original cluster client isn't needed anymore and closed.
    * @param req - Request the `ScopedClusterClient` instance will be scoped to.
+   * Supports request optionality, Legacy.Request & FakeRequest for BWC with LegacyPlatform
    */
-  public asScoped(req: { headers?: Headers } = {}) {
+  public asScoped(request?: KibanaRequest | Request | FakeRequest) {
```

Comment:
should we use `Legacy.Request` instead of `Request` to emphasise it is a deprecated api?

Reply:
> should we use Legacy.Request instead of Request to emphasise it is a deprecated api?

That makes sense to me 👍 

> request is optional request? for BWC with the current code. what is the case when 

cluster.asScoped(undefined) and ``cluster.asScoped({headers: undefined})` are valid?
Not sure I understand this issue. Both of those cases seem like they should have the same behavior. Am I missing something?

Reply:
> Not sure I understand this issue. Both of those cases seem like they should have the same behavior. Am I missing something?

ok, re-phrase: is it a valid case to perform a request to ES when authorization header is not set? I thought we always require authentication creating default `elastic`user. don't we?

Reply:
Yeah I'm not sure why `request` would ever be undefined, but the headers could be _empty_ when Security is disabled or when using OSS Kibana. I believe that case would look more like `cluster.asScoped({headers: {}})` (no `Authorization` header present).